tools/apps/src/lib.rs

```diff
@@ -686,11 +699,24 @@ impl Http3Conn {
             }
         }
 
-        path = autoindex(path, index);
+        if scheme != "http" && scheme != "https" {
+            let headers = vec![
+                quiche::h3::Header::new(":status", &"400".to_string()),
+                quiche::h3::Header::new("server", "quiche"),
+            ];
+
+            return (headers, b"Invalid scheme".to_vec());
+        }
+
+        let url = format!("{}://{}{}", scheme, host, path);
+        let url = url::Url::parse(&url).unwrap();
+
+        let mut pathbuf = path::PathBuf::from(url.path());
```

Comment:
Instead of making this mutable I think you can just do:
```rust
         let pathbuf = path::PathBuf::from(url.path());
         let pathbuf = autoindex(pathbuf, index);
```
?

Reply:
thanks, fixed